moveos/moveos-verifier/src/metadata.rs

```diff
@@ -313,25 +333,28 @@ impl<'a> ExtendedChecker<'a> {
     fn is_allowed_reference_types(&self, bt: &Type) -> bool {
         match bt {
             Type::Struct(mid, sid, _) => {
-                if self.is_allowed_input_struct(mid.qualified(*sid)) {
+                let struct_name = self
+                    .env
+                    .get_struct(mid.qualified(*sid))
+                    .get_full_name_with_address();
+                if is_allowed_input_struct(struct_name) {
                     return true;
                 }
                 false
             }
             _ => false,
         }
     }
+}
 
-    fn is_allowed_input_struct(&self, qid: QualifiedId<StructId>) -> bool {
-        let name = self.env.get_struct(qid).get_full_name_with_address();
-        matches!(
-            name.as_str(),
-            "0x1::string::String"
-                | "0x1::object_id::ObjectID"
-                | "0x1::storage_context::StorageContext"
-                | "0x1::tx_context::TxContext"
-        )
-    }
+pub fn is_allowed_input_struct(name: String) -> bool {
+    matches!(
+        name.as_str(),
+        "0x1::string::String"
+            | "0x1::object_id::ObjectID"
+            | "0x1::storage_context::StorageContext"
+            | "0x1::tx_context::TxContext"
```

Comment:
0x1::ascii::String also support.

Reply:
Okay, I will add it in a new pull request.